deps/rabbitmq_mqtt/src/rabbit_mqtt_retained_msg_store_dets.erl

```diff
@@ -2,68 +2,308 @@
 %% License, v. 2.0. If a copy of the MPL was not distributed with this
 %% file, You can obtain one at https://mozilla.org/MPL/2.0/.
 %%
-%% Copyright (c) 2007-2024 Broadcom. All Rights Reserved. The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries. All rights reserved.
+%% Copyright (c) 2007-2025 Broadcom. All Rights Reserved. The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries. All rights reserved.
 %%
 
 -module(rabbit_mqtt_retained_msg_store_dets).
 
 -behaviour(rabbit_mqtt_retained_msg_store).
 
 -include("rabbit_mqtt_packet.hrl").
+
 -include_lib("kernel/include/logger.hrl").
 
 -export([new/2, recover/2, insert/3, lookup/2, delete/2, terminate/1]).
 
--record(store_state, {table :: dets:tab_name()}).
+-record(store_state,
+        {node_table :: dets:tab_name(),    % Stores {node_id, edge_count, is_topic}
+         edge_table :: dets:tab_name(),    % Stores {{from_id, word}, to_id}
+         msg_table :: dets:tab_name(),     % Stores {node_id, topic, mqtt_msg}
+         root_id :: binary(),              % Root node ID
+         dir :: file:filename_all(),
+         vhost :: rabbit_types:vhost()}).
 
 -type store_state() :: #store_state{}.
 
 -spec new(file:name_all(), rabbit_types:vhost()) -> store_state().
 new(Dir, VHost) ->
-    {ok, TabName} = open_table(Dir, VHost),
-    #store_state{table = TabName}.
+  % delete_table_files(Dir, VHost),
```

Comment:
Not sure if tables should be cleaned first, like it is done in the ETS store.

Reply:
This code will be called on node restart. The whole idea of using DETS is to preserve that data.

I'm afraid the storage "type" and the MQTT protocol's clean session concept may be at odds here but as a general rule for DETS, we probably want to preserve its durability characteristic.

@ansd thoughts?

Reply:
> This code will be called on node restart. The whole idea of using DETS is to preserve that data.

I was so deep into the code that I missed the obvious!